### ISSUE TITLE
[feature] Link verification

### ISSUE THREAD

**Issue by filipesmedeiros** [superseriousbusiness/gotosocial#2352]
### Is your feature request related to a problem ?

It is not a problem _per se_, but it's good when you can trust the links on someone's bio :)

### Describe the solution you'd like.

Link verification just like Mastodon has.

### Describe alternatives you've considered.

I don't think there are any, that I can think of.

### Additional context.

I have filed this here https://github.com/cheeaun/phanpy/issues/316, because I thought it was a frontend issue.

It's also related to this https://github.com/superseriousbusiness/gotosocial/issues/1824, but that issue is weirdly phrased, I think?

**Comment by filipesmedeiros** [comment]
So, Mastodon does it like this: https://github.com/mastodon/mastodon/blob/0e9801443f8f91ff49b47f82151ee5984c9bd6c1/app/workers/verify_account_links_worker.rb#L12

I think they have a Sidekick worker running through links, don't know exactly when they fire the worker.

I think, since GTS is meant for small instances, we could do it synchronously when the person updates their profile (and the update includes fields that are links).

I tried to fork and create a PR, but Go is just a foreign language to me and I feel one of you would be able to finish this task much much faster! :)

If you need any help let me know

**Comment by tsmethurst** [comment]
No worries! This is one of those nice-to-haves we'll probably work on next year once we're in beta :)

**Comment by filipesmedeiros** [comment]
@daenney I agree with everything in principle, just didn't want to add complexity to the code and resource consumption for this small thing. What about a script that the admin can cron themselves? Idk

**Comment by filipesmedeiros** [comment]
Also, I don't really understand (but maybe I'm just dumb) why links from other instances aren't verified. Shouldn't that information be federated from other instances? Or does my instance have to verify _every_ link of _every_ account I'm federated with?

**Comment by daenney** [comment]
Instances are responsible for validating the profiles of their users, and may add some markup or metadata to mark them as such. However, this validation isn't a standard property that gets federated out, so other servers have no way of knowing about it.

If you look at https://docs.joinmastodon.org/spec/activitypub/#PropertyValue, you'll see how it's done. Everyone else has inherited their approach, to be compatible. As you can see, there's no attribute for it being "validated" for some definition of validation. It's a key and a value, often with HTML in it. If you look at a profile of someone on Masto who has both validated and unvalidated links in their bio and retrieve their profile, there's no difference in the `value`.

In theory there's probably ways to do it since PropertyValue has a lot of additional data it can carry according to the [schema.org definition](https://schema.org/PropertyValue). But nobody's implemented that.

**Comment by mirabilos** [comment]
@daenney I think it’s not federated out because “my own” instance is supposed to verify other profiles’ links, so I don’t have to trust “their” instance.

**Comment by daenney** [comment]
We've migrated to Codeberg. Your issue has been imported as https://codeberg.org/superseriousbusiness/gotosocial/issues/2352.
